### PR TITLE
fix(dop):  project report sql injection

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
fix(dop):  project report sql injection

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=611654&iterationID=12783&tab=BUG&type=BUG)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | Fix the bug that project report sql injection             |
| 🇨🇳 中文    |    修复了项目报表sql注入的问题          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]
`.Explain` method has SQL injection vulnerability.

**Comment by chengjoey** [comment]
> `.Explain` method has SQL injection vulnerability, it handles whole sql at client side once.

added query limition

File: internal/apps/dop/providers/project_report/query_test.go [hunk 1]
```diff
@@ -149,8 +165,8 @@ func Test_checkQueryRequest(t *testing.T) {
 			name: "invalid label operation",
 			arg: &apistructs.ProjectReportRequest{
 				OrgID: 1,
-				Start: "now-1h",
-				End:   "now",
+				Start: "2024-01-01 00:00:00",
+				End:   "2024-01-02 00:00:00",
```
Comment (sfwn) [inline internal/apps/dop/providers/project_report/query_test.go:169]: add some cases of sql injection.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/dop/providers/project_report/query_test.go (hunk 1 above)
Comment (chengjoey) [reply internal/apps/dop/providers/project_report/query_test.go:186]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**